### PR TITLE
Add actual Facebook button from SDK

### PR DESCRIPTION
**Pull request by paullin03**
Add Facebook Android SDK to project and add actual Facebook login button to the main
screen. Users will be redirected to the MainActivity after they log-in or if they are
already logged in.

### CODE REVIEW

File: frontend/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -20,7 +27,7 @@
         <activity android:name=".ViewEditListsActivity" />
         <activity android:name=".ViewOldSessionsActivity" />
         <activity android:name=".CreateNewListActivity" />
-        <activity android:name=".MainActivity"/>
+        <activity android:name=".MainActivity" />
```
Comment (richardechegaray) [inline frontend/app/src/main/AndroidManifest.xml:30]: i like the consistency with the spaces paul

File: frontend/app/src/main/java/com/quickpick/LoginActivity.java [hunk 2]
```diff
@@ -1,25 +1,104 @@
...
+        facebookLoginButton.setPermissions("email");
+        facebookLoginButton.registerCallback(facebookCallbackManager, new FacebookCallback<LoginResult>() {
+            @Override
+            public void onSuccess(LoginResult loginResult) {
+                callLogin(loginResult.getAccessToken().getToken());
+            }
+
+            @Override
+            public void onCancel() {
+                Toast.makeText(getBaseContext(), "We need FB login to work!", Toast.LENGTH_LONG).show();
```
Comment (richardechegaray) [inline frontend/app/src/main/java/com/quickpick/LoginActivity.java:51]: rly minor but for our final version we probs shouldnt use abbreviations

File: frontend/app/src/main/java/com/quickpick/payloads/LoginPayload.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.quickpick.payloads;
+
+public class LoginPayload {
+
+    private String facebookToken;
+
```
Comment (richardechegaray) [inline frontend/app/src/main/java/com/quickpick/payloads/LoginPayload.java:6]: rm whitespace

**Review by richardechegaray — COMMENTED** [review]
just looked through the 10 changed files - looked good, just some random whitespace we could get rid of in spots and some other small bits

File: frontend/app/src/main/java/com/quickpick/payloads/LoginPayload.java (hunk 3 above)
Comment (paullin03) [reply frontend/app/src/main/java/com/quickpick/payloads/LoginPayload.java:6]: Addressed, we'll use newlines between members unless they make sense to be grouped.

File: frontend/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (paullin03) [reply frontend/app/src/main/AndroidManifest.xml:30]: That's Android Studio 😅

File: frontend/app/src/main/java/com/quickpick/LoginActivity.java (hunk 2 above)
Comment (paullin03) [reply frontend/app/src/main/java/com/quickpick/LoginActivity.java]: Addressed

**Review by richardechegaray — APPROVED** [review]

**Pull request merged**